tests/conftest.py

```diff
@@ -448,3 +460,58 @@ def _make_sdata_for_testing_querying_and_aggretation() -> SpatialData:
 @pytest.fixture()
 def sdata_query_aggregation() -> SpatialData:
     return _make_sdata_for_testing_querying_and_aggretation()
+
+
+def generate_adata(n_var: int, obs: pd.DataFrame, obsm: dict[Any, Any], uns: dict[Any, Any]) -> AnnData:
+    rng = np.random.default_rng(SEED)
+    return AnnData(
+        rng.normal(size=(obs.shape[0], n_var)),
+        obs=obs,
+        obsm=obsm,
+        uns=uns,
+        dtype=np.float64,
+    )
+
+
+def _get_blobs_galaxy() -> tuple[NDArrayA, NDArrayA]:
+    blobs = data.binary_blobs(rng=SEED)
+    blobs = ndi.label(blobs)[0]
+    return blobs, data.hubble_deep_field()[: blobs.shape[0], : blobs.shape[0]]
+
+
+@pytest.fixture
+def adata_labels() -> AnnData:
+    n_var = 50
+
+    blobs, _ = _get_blobs_galaxy()
+    seg = np.unique(blobs)[1:]
+    n_obs_labels = len(seg)
+    rng = np.random.default_rng(SEED)
+
+    obs_labels = pd.DataFrame(
+        {
+            "a": rng.normal(size=(n_obs_labels,)),
+            "categorical": pd.Categorical(rng.integers(0, 2, size=(n_obs_labels,))),
+            "cell_id": pd.Categorical(seg),
+            "instance_id": range(n_obs_labels),
+            "region": ["test"] * n_obs_labels,
+        },
+        index=np.arange(n_obs_labels),
+    )
+    uns_labels = {
+        "spatial": {
+            "labels": {
+                "scalefactors": {
+                    "spot_diameter_fullres": 10,
+                    "tissue_hires_scalef": 1,
+                    "tissue_segmentation_scalef": 1,
+                }
+            }
+        },
+        "spatialdata_attrs": {"region": "test", "region_key": "region", "instance_key": "instance_id"},
+    }
+    obsm_labels = {
+        "spatial": rng.integers(0, blobs.shape[0], size=(n_obs_labels, 2)),
+        "spatial_copy": rng.integers(0, blobs.shape[0], size=(n_obs_labels, 2)),
+    }
```

Comment:
I renamed `spatial` and `spatial_copy` to `tensor` and `tensor_copy` since `spatial` is from a legacy format so this may confuse the reader.